### PR TITLE
[api-extractor] Add support for HTML tags in doc comments

### PR DESCRIPTION
**Pull request by pgonzal**
Add support for CommonMark-style HTML tags in doc comments.  This can be used to create tables `<tables>` and as a quick hack for other features such as code ranges (`<code>`) or boldface (`<b>`) until the TSDoc parser is integrated.

To avoid parsing ambiguities with other AEDoc constructs, API Extractor will ensure that this string is a complete and properly formatted opening or closing HTML tag such as `<td>` or `</td>` or ``.  Beyond this, API Extractor does NOT attempt to parse the tag attributes, or verify that opening/closing pairs are balanced, or determine whether the nested tree is valid HTML.  That responsibility is left to the consuming documentation engine.

If you want a literal HTML tag, it can be escaped using a CommonMark backslash:

```javascript
/**
 * This is shown as a literal "\<br />" tag whereas this "<br />" would
 * get rendered as an actual paragraph break.
 */
```

### CODE REVIEW

**Comment by pgonzal** [comment]
@AlexJerabek FYI

File: apps/api-extractor/src/aedoc/DocElementParser.ts [hunk 1]
```diff
@@ -17,6 +17,18 @@ import { ResolvedApiItem } from '../ResolvedApiItem';
...
+   * Matches one of:
+   * - an escape sequence, i.e. backslash followed by a non-alphabetical character
+   * - an HTML opening tag such as `<td>` or or `<img src="example.gif" />`
+   * - an HTML closing tag such `</td>`
+   *
+   * Note that the greedy nature of the RegExp ensures that `\<td>` will get interpreted
+   * as an escaped "<", whereas `\\<td>` will get interpreted as an escaped backslash
+   * followed by an HTML element.
+   */
+  private static _htmlTagRegExp: RegExp = /\\[^a-zA-Z\s]|<[\w\-]+(?:\s+[\w\-]+\s*=\s*"[^"]*")*\s*\/?>|<\/[\w\-]+>/g;
```
Comment (iclanton) [inline apps/api-extractor/src/aedoc/DocElementParser.ts]: >\\[^a-zA-Z\s]|<[\w\-]+(?:\s+[\w\-]+\s*=\s*"[^"]*")*\s*\/?>|<\/[\w\-]+> [](start = 43, length = 70)

Add support for single quotes as well.

File: apps/api-extractor/src/aedoc/test/DocElementParser.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+function makeSnapshot(s: string): void {
+  expect(DocElementParser.parseMarkdownishText(s)).toMatchSnapshot();
+}
+
+test('DocElementParser.parseMarkdownishText', () => {
+  makeSnapshot('not an escape:  C:\\Dog\\Cat');
+  makeSnapshot('escaped symbols: \\$\\^');
+  makeSnapshot('escaped backslashes: \\\\ \\\\\\ end');
+  makeSnapshot('escaped HTML: \\<td>');
+  makeSnapshot('real HTML: <td> </td> <a href="#" /a>');
```
Comment (iclanton) [inline apps/api-extractor/src/aedoc/test/DocElementParser.test.ts]: >a [](start = 51, length = 1)

Did you intend this to be valid? Drop the `a` at the end to fix it.

Comment (iclanton) [inline apps/api-extractor/src/aedoc/test/DocElementParser.test.ts]: >makeSnapshot [](start = 2, length = 12)

Maybe add a case with both escaped and unescaped HTML.

Comment (iclanton) [inline apps/api-extractor/src/aedoc/test/DocElementParser.test.ts]: After you fix the regex, add a a case with single-quoted attributes.

**Pull request merged**